storage/innobase/fil/fil0fil.cc

```diff
@@ -11708,10 +11710,10 @@ dberr_t Tablespace_dirs::scan(bool populate_fil_cache) {
 
   if (err == DB_SUCCESS && populate_fil_cache) {
     bool result = true;
-    std::function<void(const Const_iter &, const Const_iter &, size_t)> open =
-        std::bind(&Tablespace_dirs::open_ibd, this, _1, _2, _3, result);
+    std::function<void(const Const_iter &, const Const_iter &, size_t, bool &)>
```

Comment:
Do you know what happens when such a file is present and mysqld will abort or start?

Reply:
It will abort:
```2020-11-13T17:14:19.925432Z 1 [Note] [MY-013083] [InnoDB] Log background threads are being started...
2020-11-13T17:14:19.925963Z 1 [Note] [MY-012532] [InnoDB] Applying a batch of 62 redo log records ...
2020-11-13T17:14:19.928883Z 1 [Note] [MY-012533] [InnoDB] 100%
2020-11-13T17:14:20.429197Z 1 [Note] [MY-012535] [InnoDB] Apply batch completed!
2020-11-13T17:14:20.429693Z 1 [ERROR] [MY-012179] [InnoDB] Could not find any file associated with the tablespace ID: 3
2020-11-13T17:14:20.430048Z 1 [ERROR] [MY-012964] [InnoDB] Use --innodb-directories to find the tablespace files. If that fails then use --innodb-force-recovery=1 to ignore this and to permanently lose all changes to the missing tablespace(s)
2020-11-13T17:14:20.430548Z 1 [ERROR] [MY-012930] [InnoDB] Plugin initialization aborted at srv0start.cc[2864] with error Generic error.
2020-11-13T17:14:24.874984Z 1 [ERROR] [MY-010334] [Server] Failed to initialize DD Storage Engine
2020-11-13T17:14:24.875845Z 0 [ERROR] [MY-010020] [Server] Data Dictionary initialization failed.
2020-11-13T17:14:24.876616Z 0 [ERROR] [MY-010119] [Server] Aborting
```